Apps/WebClient/src/NewClientApp/src/components/private/dependent/AddDependentComponent.vue

```diff
@@ -0,0 +1,445 @@
+<script setup lang="ts">
+import { useVuelidate } from "@vuelidate/core";
+import { helpers } from "@vuelidate/validators";
+import { minLength, required, sameAs } from "@vuelidate/validators";
+import { Duration } from "luxon";
+import { vMaska } from "maska";
+import { computed, nextTick, ref, watch } from "vue";
+
+import HgButtonComponent from "@/components/common/HgButtonComponent.vue";
+import HgDatePickerComponent from "@/components/common/HgDatePickerComponent.vue";
+import HgIconButtonComponent from "@/components/common/HgIconButtonComponent.vue";
+import LoadingComponent from "@/components/common/LoadingComponent.vue";
+import TooManyRequestsComponent from "@/components/error/TooManyRequestsComponent.vue";
+import { ActionType } from "@/constants/actionType";
+import { Loader } from "@/constants/loader";
+import { container } from "@/ioc/container";
+import { SERVICE_IDENTIFIER } from "@/ioc/identifier";
+import AddDependentRequest from "@/models/addDependentRequest";
+import { DateWrapper, IDateWrapper } from "@/models/dateWrapper";
+import { ResultError } from "@/models/errors";
+import { IDependentService } from "@/services/interfaces";
+import { useConfigStore } from "@/stores/config";
+import { useDependentStore } from "@/stores/dependent";
+import { useErrorStore } from "@/stores/error";
+import { useLoadingStore } from "@/stores/loading";
+import { useUserStore } from "@/stores/user";
+import { phnMask } from "@/utility/masks";
+import ValidationUtil from "@/utility/validationUtil";
+
+const emit = defineEmits<{
+    (e: "handle-submit"): void;
+}>();
+
+interface props {
+    disabled?: boolean;
+}
+
+withDefaults(defineProps<props>(), {
+    disabled: false,
+});
+
+defineExpose({
+    hideDialog,
+});
+
+const emptyDependent = {
+    firstName: "",
+    lastName: "",
+    dateOfBirth: "",
+    PHN: "",
+};
+
+const phnMaskaOptions = {
+    mask: phnMask,
+    eager: true,
+};
+
+const maxBirthdate = new DateWrapper();
+
+const dependentService = container.get<IDependentService>(
+    SERVICE_IDENTIFIER.DependentService
+);
+
+const configStore = useConfigStore();
+const dependentStore = useDependentStore();
+const errorStore = useErrorStore();
+const loadingStore = useLoadingStore();
+const userStore = useUserStore();
+
+const isVisible = ref(false);
+const errorMessage = ref("");
+const errorType = ref<ActionType | null>(null);
+const accepted = ref(false);
+const dependent = ref<AddDependentRequest>(emptyDependent);
```

Comment:
This should be a copy of `emptyDependent`, so that we don't mutate the `emptyDependent` values
```suggestion
const dependent = ref<AddDependentRequest>({ ...emptyDependent });
```